meta-mbl-distro/recipes-support/swupdate/files/defconfig

```diff
@@ -0,0 +1,98 @@
+#
+# Automatically generated file; DO NOT EDIT.
+# Swupdate Configuration
+#
+CONFIG_HAVE_DOT_CONFIG=y
+
+#
+# Swupdate Settings
+#
+
+#
+# General Configuration
+#
+CONFIG_SCRIPTS=y
+CONFIG_HW_COMPATIBILITY=y
+CONFIG_HW_COMPATIBILITY_FILE="/etc/hwrevision"
+CONFIG_SW_VERSIONS_FILE="/etc/sw-versions"
```

Comment:
I'm not convinced swupdate's sw compatibility feature will be suitable for us... Can we turn it off? If we have to keep this file then it probably should live in our user config partition.

Reply:
Testing it works with this switched off now.

Reply:
I don't think we can actually turn this off, we can just specify the file it uses. If the file doesn't exist it looks like swupdate's `artifact_versions.c` creates it at `/etc/swversions` https://github.com/sbabic/swupdate/blob/f46e9e1ff4617aa876c9ede964615a8e806bca83/corelib/artifacts_versions.c

I'll try moving it to the user config partition.

Reply:
Mmm, okay. I suppose we'll just have to live with it in the user config partition.

Reply:
I've had to hard-code the /user/config path in the `defconfig`. I tried to use `mbl-var-placeholders` but that wouldn't work as the task was running after the `swupdate` compilation task so the changes weren't being picked up.

Reply:
Would something like the following work?
```
do_configure_prepend() {
    printf 'CONFIG_SW_VERSIONS_FILE="%s"' "${MBL_CONFIG_DIR}/sw-versions" >> "${WORKDIR}/defconfig"
}
```

Reply:
Hmm... yeah... I should've thought of that! I'll try it.